src/DataStructures/Tensor/EagerMath/Magnitude.hpp

```diff
@@ -48,11 +50,51 @@ Scalar<DataType> magnitude(
 /// \details
 /// Computes the square root of the absolute value of the scalar.
 template <typename DataType>
-Scalar<DataType> sqrt_magnitude(
-    const Scalar<DataType>& input) noexcept {
+Scalar<DataType> sqrt_magnitude(const Scalar<DataType>& input) noexcept {
   return Scalar<DataType>{sqrt(abs(get(input)))};
 }
 
+namespace L2Norm_detail {
+template <typename DataType, typename Symm, typename IndexList>
+Scalar<DataType> pointwise_l2_norm_square(
+    const Tensor<DataType, Symm, IndexList>& tensor) noexcept {
+  auto pointwise_l2_normsq = make_with_value<DataType>(tensor, 0.);
+  for (auto it = tensor.begin(); it != tensor.end(); ++it) {
+    // In order to handle tensory symmetries, we multiply the square of each
+    // stored component with its multiplicity
+    pointwise_l2_normsq += tensor.multiplicity(it) * square(*it);
+  }
+  return Scalar<DataType>{pointwise_l2_normsq};
```

Comment:
I think this copies the DataVector, so better `std::move(pointwise_l2_normsq)` or `make_with_value<Scalar<DataType>>` to begin with?

Reply:
done